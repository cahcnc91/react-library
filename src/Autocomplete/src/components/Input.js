import React from "react";
import withStyles from "react-jss";

const styles = {
  showResultsBottom: {
    width: "100%",
    padding: "0.1rem",
    borderBottom: "1px solid #ced4da",
    boxShadow: "none",
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    height: "100%"
  },
  noShowResults: {
    width: "100%"
  },
  flexRow: {
    display: "flex",
    flexDirection: "row",
    width: "100%",
    alignItems: "center",
    justifyContent: "space-between"
  },
  divider: {
    backgroundColor: "#AEAEAE",
    width: "1px",
    height: "1.5rem"
  },
  icon: {
    marginRight: "10px",
    fontSize: "1.6rem",
    color: "#AEAEAE",
    cursor: "pointer"
  },
  input: {
    width: "calc(100% - 45px)",
    border: "none",
    borderRadius: "8px",
    padding: ".8rem 0 .8rem .7rem",
    fontSize: ".9rem",
    "&:focus": {
      outline: "none"
    },
    "&:placeholder": {
      fontSize: "1.0rem"
    }
  },
  flexRow1: {
    display: "flex",
    flexDirection: "row",
    width: "45px",
    alignItems: "center",
    justifyContent: "space-between"
  },
};

/** universal-search-bar-input */
const Input = ({
  showResults,
  searchString,
  toggleShowResults,
  onKeyDown,
  filterResults,
  classes
}) => {

  return (
    <div
      className={
        showResults ? classes.showResultsBottom : classes.noShowResults
      }
    >
      <div className={classes.flexRow}>
        <input
          id="searchBarInput"
          value={searchString}
          type="text"
          autoComplete="off"
          placeholder={"Search for..."}
          onChange={e => filterResults(e.target.value)}
          className={classes.input}
          onClick={toggleShowResults}
          onKeyDown={onKeyDown}
        />
        <div className={classes.flexRow1}>
          <div className={classes.divider} />
          {searchString === "" ? (
              <div>search</div>
            // <Search className={classes.icon} />
          ) : (
            // <Clear className={classes.icon} onClick={() => filterResults("")} />
            <div>clear</div>
          )}
        </div>
      </div>
    </div>
  );
};

export default withStyles(styles)(Input);
