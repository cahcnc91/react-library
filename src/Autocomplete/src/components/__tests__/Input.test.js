import React from "react";
import { mount } from "enzyme";
import Input from "../Input";

const toggleShowResults = () => {};
const onKeyDown = () => {};
const onChange = () => {};

describe("Input", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <Input
        showResults={true}
        searchString={"Java"}
        toggleShowResults={toggleShowResults}
        onKeyDown={onKeyDown}
        onChange={onChange}
      />
    );
  });

  test("snapshot renders", () => {
    let tree = wrapper.html();
    expect(tree).toMatchSnapshot();
  });

  it("renders input, input value and class", () => {
    expect(wrapper.find('input[type="text"]').props().value).toEqual("Java");
    expect(wrapper.find('input[type="text"]').props().className).toEqual("Input-input-0-2-7");
  });
});
