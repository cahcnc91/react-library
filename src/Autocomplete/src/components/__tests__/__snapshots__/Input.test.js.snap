// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot renders 1`] = `"<div class=\\"Input-showResultsBottom-0-2-1\\"><div class=\\"Input-flexRow-0-2-3\\"><input id=\\"searchBarInput\\" type=\\"text\\" autocomplete=\\"off\\" placeholder=\\"Search for...\\" class=\\"Input-input-0-2-7\\" value=\\"Java\\"><div class=\\"Input-flexRow1-0-2-8\\"><div class=\\"Input-divider-0-2-4\\"></div><svg id=\\"Capa_1\\" enable-background=\\"new 0 0 413.348 413.348\\" height=\\"512\\" viewBox=\\"0 0 413.348 413.348\\" width=\\"512\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"Input-iconClear-0-2-6\\"><path d=\\"m413.348 24.354-24.354-24.354-182.32 182.32-182.32-182.32-24.354 24.354 182.32 182.32-182.32 182.32 24.354 24.354 182.32-182.32 182.32 182.32 24.354-24.354-182.32-182.32z\\"></path></svg></div></div></div>"`;
